### PR TITLE
WE-829 Implement DeleteObjectsWorker

### PR DESCRIPTION
**Pull request by janburak**
**blocked by WE-828 and WE-827**

## Fixes
This pull request fixes WE-829

## Description
Implement DeleteObjectsWorker to replace existing delete workers.

## Type of change

* Enhancement of existing functionality

## Impacted Areas in Application

* API, Frontend

## Checklist:

*Communication*
* [x] PR is related to an issue
* [ ] I have made corresponding changes to the documentation

*Code quality*
* [x] Code follows the style guidelines
* [x] I have self-reviewed my code
* [x] No new warnings are generated

*Test coverage*
* [x] Existing tests pass
* [x] New code is covered by passing tests

### CODE REVIEW

File: Src/WitsmlExplorer.Api/Models/EntityType.cs [hunk 1]
```diff
@@ -35,5 +40,24 @@ public class EntityTypeHelper
                    return lower.Last() == 'y' ? lower.Remove(lower.Length - 1) + "ies" : lower + "s";
                });
         }
+
+        public static Witsml.Data.ObjectOnWellbore EntityTypeToObjectOnWellbore(EntityType type)
```
Comment (HakonHesselberg) [inline Src/WitsmlExplorer.Api/Models/EntityType.cs]: Should the class EntityTypeHelper be static?

File: Src/WitsmlExplorer.Api/Workers/Delete/DeleteObjectsWorker.cs [hunk 2]
```diff
@@ -74,5 +83,6 @@ public async Task<(WorkerResult, RefreshAction)> DeleteObjectsOnWellbore(IWitsml
                 ? (new WorkerResult(witsmlClient.GetServerHostname(), true, successString), refreshAction)
                 : (new WorkerResult(witsmlClient.GetServerHostname(), false, $"{successString} Failed to delete some {queries.First().GetType().Name}s", errorReason, null), successUids.Count > 0 ? refreshAction : null);
         }
+
```
Comment (HakonHesselberg) [inline Src/WitsmlExplorer.Api/Workers/Delete/DeleteObjectsWorker.cs]: Extra space

File: Src/WitsmlExplorer.Api/Workers/Delete/DeleteObjectsWorker.cs [hunk 3]
```diff
@@ -42,13 +51,13 @@ public async Task<(WorkerResult, RefreshAction)> DeleteObjectsOnWellbore(IWitsml
                     QueryResult result = await witsmlClient.DeleteFromStoreAsync(query.AsSingletonWitsmlList());
```
Comment (HakonHesselberg) [inline Src/WitsmlExplorer.Api/Workers/Delete/DeleteObjectsWorker.cs:52]: Line 47: results is never used

File: Src/WitsmlExplorer.Api/Workers/Delete/DeleteObjectsWorker.cs [hunk 4]
```diff
@@ -74,5 +83,6 @@ public async Task<(WorkerResult, RefreshAction)> DeleteObjectsOnWellbore(IWitsml
                 ? (new WorkerResult(witsmlClient.GetServerHostname(), true, successString), refreshAction)
                 : (new WorkerResult(witsmlClient.GetServerHostname(), false, $"{successString} Failed to delete some {queries.First().GetType().Name}s", errorReason, null), successUids.Count > 0 ? refreshAction : null);
```
Comment (HakonHesselberg) [inline Src/WitsmlExplorer.Api/Workers/Delete/DeleteObjectsWorker.cs]: null is already the default value

File: Tests/WitsmlExplorer.IntegrationTests/Api/Workers/DeleteObjectsWorkerTests.cs [hunk 5]
```diff
@@ -7,40 +7,42 @@
...
                 {
-                    WellUid = "8c77de13-4fad-4b2e-ba3d-7e6b0e35a394",
-                    WellboreUid = "44e7a064-c2f2-4a3a-9259-5ab92085e110",
-                    ObjectUids = new string[] { "2YA2M49" }
+                    WellUid = "<well_uid>",
+                    WellboreUid = "<wellbore_uid>",
+                    ObjectUids = new string[] { "<object_uid_1>", "<object_uid_2>" },
+                    ObjectType = EntityType.Log // set to the type of object you want to delete
                 }
+
```
Comment (HakonHesselberg) [inline Tests/WitsmlExplorer.IntegrationTests/Api/Workers/DeleteObjectsWorkerTests.cs]: Extra space

**Review by HakonHesselberg — DISMISSED** [review]

File: Src/WitsmlExplorer.Api/Workers/Delete/DeleteObjectsWorker.cs (hunk 4 above)
Comment (janburak) [reply Src/WitsmlExplorer.Api/Workers/Delete/DeleteObjectsWorker.cs]: Is there any better way to write this line?

**Review by steinsiv — APPROVED** [review]
```js
- [x] Read issue and PR
- [x] Pulled branch and manually tested
- [x] Verify that PR resolves issue
- [x] Reviewed the code
```

**Pull request merged**